src/Purebred/UI/Actions.hs

```diff
@@ -1015,23 +1015,49 @@ scrollPageDown = Action
   , _aAction = Brick.vScrollPage (makeViewportScroller @ctx) T.Down
   }
 
-scrollNextWord :: forall ctx v. (Scrollable ctx) => Action v ctx ()
-scrollNextWord =
+scrollNextMatch :: forall ctx v. (Scrollable ctx) => Action v ctx ()
+scrollNextMatch =
   Action
-    { _aDescription = ["find next word in mail body"]
-    , _aAction = do
-        Brick.vScrollToBeginning (makeViewportScroller @ctx)
-        b <- gets (has (asMailView . mvScrollSteps))
-        if b
-          then do
-            modifying (asMailView . mvScrollSteps) Brick.focusNext
-            nextLine <- preuse (asMailView . mvScrollSteps . to Brick.focusGetCurrent . _Just . _2)
-            let scrollBy = view (non 0) nextLine
-            Brick.vScrollBy (makeViewportScroller @ctx) scrollBy
-          else
-            showWarning "No match"
+    { _aDescription = ["scroll to next match in mail body"]
+    , _aAction = scrollMatch (makeViewportScroller @ctx) (+1)
+    }
+
+scrollPreviousMatch :: forall ctx v. (Scrollable ctx) => Action v ctx ()
+scrollPreviousMatch =
+  Action
+    { _aDescription = ["scroll to previous match in mail body"]
+    , _aAction = scrollMatch (makeViewportScroller @ctx) (subtract 1)
     }
 
+-- | Scroll to a particular match, the index of which can be calculated
+-- from the current index.  So:
```

Comment:
This basically scrolls relatively to a match if I read this correctly instead of what I used to be relative to the top (e.g. line 0)?

Reply:
Under the hood, it still does "scroll to beginning, then scroll forward <n> lines".  But now we calculate <n> a bit differently.